internal/testrunner/runners/system/tester.go

```diff
@@ -556,13 +602,12 @@ func (r *tester) runTestPerVariant(ctx context.Context, result *testrunner.Resul
 	if err != nil {
 		return nil, fmt.Errorf("unable to load system test case file '%s': %w", configFile, err)
 	}
-	logger.Debugf("Using config: %q", testConfig.Name())
 
 	partial, err := r.runTest(ctx, testConfig, svcInfo)
 
 	tdErr := r.tearDownTest(ctx)
 	if err != nil {
-		return partial, err
+		return partial, errors.Join(err, fmt.Errorf("failed to tear down runner: %w", tdErr))
```

Comment:
Not able yet to show the teardown error when the context was cancelled (pressing Ctrl+C). 

As it was previously, deleting test policies handler was failing but the error did not raised

```
^C2024/06/13 19:32:38 DEBUG deleting test policies...
2024/06/13 19:32:38  INFO Signal caught!
2024/06/13 19:32:38 DEBUG POST https://127.0.0.1:5601/api/fleet/agent_policies/delete
2024/06/13 19:32:39 DEBUG POST https://127.0.0.1:5601/api/fleet/agent_policies/delete
2024/06/13 19:32:39 DEBUG Failed deleting test policy handler...: error cleaning up test policy: could not delete policy; API status code = 400; response body = {"statusCode":400,"error":"Bad Request","message":"Cannot delete an agent policy that is assigned to any active or inactive agents"}
2024/06/13 19:32:39 ERROR context error: context canceled
2024/06/13 19:32:39 ERROR context error: context canceled
2024/06/13 19:32:39 ERROR context error: context canceled
2024/06/13 19:32:39 DEBUG Uninstalling package...
2024/06/13 19:32:39 DEBUG GET https://127.0.0.1:5601/api/fleet/epm/packages/nginx
2024/06/13 19:32:40 DEBUG DELETE https://127.0.0.1:5601/api/fleet/epm/packages/nginx/999.999.999
interrupted
Command exited with non-zero status 130
```

Reply:
Ok, understood. I'll remove then this `errors.Join`

Thanks!